PMM-7 pull upstream changes

BupycHuk opened the pull request.

Eric-Tyrrell22 committed:
Adds 1kB and 2kB units (#915)

Signed-off-by: Eric tyrrell <eric.tyrrell18+github@gmail.com>

sysadmind committed:
Add error log when probe collector creation fails (#918)

Signed-off-by: Joe Adams <github@joeadams.io>

dswarbrick committed:
Fix test build failures on 32-bit arch again (#919)

Another case of untyped integer overflows on 32-bit arch.

Signed-off-by: Daniel Swarbrick <daniel.swarbrick@gmail.com>

SuperQ committed:
Add 32-bit testing to CI (#920)

Run Go tests with 32-bit to validate value overflow.

Signed-off-by: SuperQ <superq@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.16.0 to 1.17.0 (#925)

* Bump github.com/prometheus/client_golang from 1.16.0 to 1.17.0

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.16.0 to 1.17.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.16.0...v1.17.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update tests for latest client_golang.

Signed-off-by: SuperQ <superq@gmail.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: SuperQ <superq@gmail.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: SuperQ <superq@gmail.com>

prombot committed:
Update common Prometheus files (#926)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

sysadmind committed:
Adjust collector to use separate connection per scrape (#931)

Fixes #921

Signed-off-by: Joe Adams <github@joeadams.io>

dependabot[bot] committed:
Bump golang.org/x/net from 0.10.0 to 0.17.0 (#936)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.10.0 to 0.17.0.
- [Commits](https://github.com/golang/net/compare/v0.10.0...v0.17.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

SuperQ committed:
Release v0.15.0 (#944)

* [ENHANCEMENT] Add 1kB and 2kB units #915
* [BUGFIX] Add error log when probe collector creation fails #918
* [BUGFIX] Fix test build failures on 32-bit arch #919
* [BUGFIX] Adjust collector to use separate connection per scrape #936

Signed-off-by: SuperQ <superq@gmail.com>

prombot committed:
Update common Prometheus files (#951)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/not_signed)](https://cla-assistant.percona.com:443/percona/postgres_exporter?pullRequest=171) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/postgres_exporter?pullRequest=171) before we can accept your contribution.<br/>**0** out of **8** committers have signed the CLA.<br/><br/>:x: sysadmind<br/>:x: avances123<br/>:x: pierresouchay<br/>:x: cristiangreco<br/>:x: dannotripp<br/>:x: Nudin<br/>:x: prombot<br/>:x: eugercek<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/postgres_exporter?pullRequest=171) it.</sub>

prombot committed:
Update common Prometheus files (#963)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

shamil committed:
pg_replication_slot: add slot type label (#960)

Signed-off-by: Alex Simenduev <shamil.si@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.44.0 to 0.45.0 (#948)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.44.0 to 0.45.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Commits](https://github.com/prometheus/common/compare/v0.44.0...v0.45.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_model (#949)

Bumps [github.com/prometheus/client_model](https://github.com/prometheus/client_model) from 0.4.1-0.20230718164431-9a2bf3000d16 to 0.5.0.
- [Release notes](https://github.com/prometheus/client_model/releases)
- [Commits](https://github.com/prometheus/client_model/commits/v0.5.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_model
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sitole committed:
pg_stat_database: added support for `active_time` counter (#961)

* feat(pg_stat_database): active time metric

---------

Signed-off-by: Jiri Sveceny <jiri.sveceny@icloud.com>

dependabot[bot] committed:
Bump golang.org/x/crypto from 0.14.0 to 0.17.0 (#988)

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.14.0 to 0.17.0.
- [Commits](https://github.com/golang/crypto/compare/v0.14.0...v0.17.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.17.0 to 1.18.0 (#993)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.17.0 to 1.18.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.17.0...v1.18.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

keegancsmith committed:
use Info level for excluded databases log message (#1003)

This is the only log message which didn't specify a level in the
postgres_exporter. I am unsure if this log message should be info or
debug, but leaning towards the more important since previously it would
just always log.

The way I validated this was the only non-leveled logger was via grep.
Both of these only returned this callsite previously:

  git grep 'logger\.Log'
  git grep '\.Log(' | grep -v level

Signed-off-by: Keegan Carruthers-Smith <keegan.csmith@gmail.com>

jocelynthode committed:
Add connection limits metrics for pg_roles and pg_database (#997)

* Add database connection limits metrics

Signed-off-by: Jocelyn Thode <jocelyn@thode.email>

* Add roles connection limits metrics

Signed-off-by: Jocelyn Thode <jocelyn@thode.email>

* Fix copyright year

Co-authored-by: Joe Adams <github@joeadams.io>
Signed-off-by: Jocelyn Thode <jocelynthode@users.noreply.github.com>

* Fix spacing in pgDatabaseQuery

Co-authored-by: Joe Adams <github@joeadams.io>
Signed-off-by: Jocelyn Thode <jocelynthode@users.noreply.github.com>

* Fix case on pgRolesConnectionLimitsQuery

Co-authored-by: Joe Adams <github@joeadams.io>
Signed-off-by: Jocelyn Thode <jocelynthode@users.noreply.github.com>

* Do not add roleMetrics when row is not valid

Signed-off-by: Jocelyn Thode <jocelyn@thode.email>

---------

Signed-off-by: Jocelyn Thode <jocelyn@thode.email>
Signed-off-by: Jocelyn Thode <jocelynthode@users.noreply.github.com>
Co-authored-by: Joe Adams <github@joeadams.io>

dependabot[bot] committed:
Bump github.com/DATA-DOG/go-sqlmock from 1.5.0 to 1.5.2 (#1000)

Bumps [github.com/DATA-DOG/go-sqlmock](https://github.com/DATA-DOG/go-sqlmock) from 1.5.0 to 1.5.2.
- [Release notes](https://github.com/DATA-DOG/go-sqlmock/releases)
- [Commits](https://github.com/DATA-DOG/go-sqlmock/compare/v1.5.0...v1.5.2)

---
updated-dependencies:
- dependency-name: github.com/DATA-DOG/go-sqlmock
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.18.0 to 1.19.0 (#1011)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.18.0 to 1.19.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/v1.19.0/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.18.0...v1.19.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_model from 0.5.0 to 0.6.0 (#1010)

Bumps [github.com/prometheus/client_model](https://github.com/prometheus/client_model) from 0.5.0 to 0.6.0.
- [Release notes](https://github.com/prometheus/client_model/releases)
- [Commits](https://github.com/prometheus/client_model/compare/v0.5.0...v0.6.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_model
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump google.golang.org/protobuf from 1.32.0 to 1.33.0 (#1014)

Bumps google.golang.org/protobuf from 1.32.0 to 1.33.0.

---
updated-dependencies:
- dependency-name: google.golang.org/protobuf
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.10.0 to 0.11.0 (#992)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.10.0 to 0.11.0.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.10.0...v0.11.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump golang.org/x/net from 0.20.0 to 0.23.0 (#1021)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.20.0 to 0.23.0.
- [Commits](https://github.com/golang/net/compare/v0.20.0...v0.23.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

MarcWort committed:
feat: Add safe_wal_size and wal_status to replication_slot  (#1027)

* feat: Add safe_wal_size to replication_slot

Signed-off-by: MarcWort <113890636+MarcWort@users.noreply.github.com>

* feat: Add wal_status to replication_slot

Signed-off-by: MarcWort <113890636+MarcWort@users.noreply.github.com>

---------

Signed-off-by: MarcWort <113890636+MarcWort@users.noreply.github.com>

MarcWort committed:
fix: Only query active_time on pg>=14 (#1045)

Signed-off-by: MarcWort <113890636+MarcWort@users.noreply.github.com>

fhackenberger committed:
Update README.md (#1038)

Better example for the quick start with prometheus config and avoiding deprecated env variables.

Signed-off-by: fhackenberger <florian@hackenberger.at>

saz committed:
stop logging errors on replicas, fixes #547 (#1048)

Signed-off-by: Steffen Zieger <github@saz.sh>

prombot committed:
Update common Prometheus files (#983)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#1076)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

tjhop committed:
chore!: adopt log/slog, drop go-kit/log (#1073)

* ci: update go to version 1.23

Signed-off-by: TJ Hoplock <t.hoplock@gmail.com>

* build(deps): bump prometheus/{client_golang,common,exporter-toolkit}

Signed-off-by: TJ Hoplock <t.hoplock@gmail.com>

* chore!: adopt log/slog, drop go-kit/log

The bulk of this change set was automated by the following script which
is being used to aid in converting the various exporters/projects to use
slog:

https://gist.github.com/tjhop/49f96fb7ebbe55b12deee0b0312d8434

Signed-off-by: TJ Hoplock <t.hoplock@gmail.com>

---------

Signed-off-by: TJ Hoplock <t.hoplock@gmail.com>
Co-authored-by: Ben Kochie <superq@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.20.4 to 1.20.5 (#1079)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.20.4 to 1.20.5.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.20.4...v1.20.5)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.60.0 to 0.60.1 (#1080)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.60.0 to 0.60.1.
- [Release notes](https://github.com/prometheus/common/releases)
- [Changelog](https://github.com/prometheus/common/blob/main/RELEASE.md)
- [Commits](https://github.com/prometheus/common/compare/v0.60.0...v0.60.1)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

bitfehler committed:
Make walreceiver collector useful w/o repmgr (#1086)

In a streaming replication setup that was created without replication
manager (`repmgr`), the `stat_wal_receiver` collector does not return
any metrics, because one value it wants to export is not present.

This is rather overly opinionated. The missing metric is comparatively
uninteresting and does not justify discarding all the others. And
replication setups created without `repmgr` are not exactly rare.

This commit makes the one relevant metric optional and simply skips it
if the respective value cannot be determined.

Signed-off-by: Conrad Hoffmann <ch@bitfehler.net>

prombot committed:
Update common Prometheus files (#1083)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.13.0 to 0.13.1 (#1081)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.13.0 to 0.13.1.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.13.0...v0.13.1)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#1087)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

sysadmind committed:
Update changelog and version for a v0.16.0 release (#1088)

Signed-off-by: Joe Adams <github@joeadams.io>

sysadmind committed:
Fix version header in changelog (#1089)

Signed-off-by: Joe Adams <github@joeadams.io>

jyothikirant-sayukth committed:
Update pg_long_running_transactions.go (#1092)

To extract time in seconds for pg_long_running_transactions_oldest_timestamp_seconds query which currently return epoch time.

Signed-off-by: Jyothi Kiran Thammana <147131742+jyothikirant-sayukth@users.noreply.github.com>

aagarwalla-fx committed:
Fix to replace dashes with underscore in the metric names (#1103)

* Fix to replace dashes with underscore in the metric names

Signed-off-by: aagarwalla-fx <arpit.agarwalla@falconx.io>

* Code style fix

Signed-off-by: aagarwalla-fx <arpit.agarwalla@falconx.io>

---------

Signed-off-by: aagarwalla-fx <arpit.agarwalla@falconx.io>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.13.1 to 0.13.2 (#1108)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.13.1 to 0.13.2.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.13.1...v0.13.2)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

n-rodriguez committed:
Checkpoint related columns in PG 17 have been moved from pg_stat_bgwriter to pg_stat_checkpointer (#1072)

* Checkpoint related columns in PG 17 have been moved from pg_stat_bgwriter to pg_stat_checkpointer

Fix https://github.com/prometheus-community/postgres_exporter/issues/1060

See: https://www.dbi-services.com/blog/postgresql-17-new-catalog-view-pg_stat_checkpointer/
Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>

* Add support for pg_stat_checkpointer

See: https://www.dbi-services.com/blog/postgresql-17-new-catalog-view-pg_stat_checkpointer/
Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>

* Run integration tests with Postgres 17

Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>

* Update date in file header

Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>

---------

Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>

prombot committed:
Update common Prometheus files (#1090)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#1110)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

khiemdoan committed:
Add Postgres 17 for CI test (#1105)

Signed-off-by: Khiem Doan <doankhiem.crazy@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.61.0 to 0.62.0 (#1118)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.61.0 to 0.62.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Changelog](https://github.com/prometheus/common/blob/main/RELEASE.md)
- [Commits](https://github.com/prometheus/common/compare/v0.61.0...v0.62.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#1124)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

NevermindZ4 committed:
pg_stat_statements PG17 (#1114)

Signed-off-by: Nevermind <79126473+NevermindZ4@users.noreply.github.com>

michael-todorovic committed:
fix: handle pg_replication_slots on pg<13 (#1098)

* fix: handle pg_replication_slots on pg<13

Signed-off-by: Michael Todorovic <michael.todorovic@outlook.com>

* fix: tests

Signed-off-by: Michael Todorovic <michael.todorovic@outlook.com>

---------

Signed-off-by: Michael Todorovic <michael.todorovic@outlook.com>

fgalind1 committed:
feat: add wait/backend to pg_stat_activity (#1106)

Signed-off-by: Felipe Galindo Sanchez <felipe.galindo.sanchez@intel.com>

bitfehler committed:
Export last replay age in replication collector (#1085)

The exported replication lag does not handle all failure modes, and can
report 0 for replicas that are out of sync and incapable of recovery.

A proper replacement for that metric would require a different approach
(see e.g. #1007), but for a lot of folks, simply exporting the age of
the last replay can provide a pretty strong signal for something being
amiss.

I think this solution might be preferable to #977, though the lag
metric needs to be fixed or abandoned eventually.

Signed-off-by: Conrad Hoffmann <ch@bitfehler.net>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.13.2 to 0.14.0 (#1126)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.13.2 to 0.14.0.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.13.2...v0.14.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sysadmind committed:
Fix missing dsn sanitization for logging (#1104)

This log line was not sanitized previously which could result in logging sensitive information. I have scanned the rest of the files and I don't see anywhere else that DSN is used in a log line without this filter.

Resolves #1042

Signed-off-by: Joe Adams <github@joeadams.io>

n-rodriguez committed:
Skip pg_stat_checkpointer collector if pg<17 (#1112)

* fix: skip collector if pg<17

Signed-off-by: Michael Todorovic <michael.todorovic@outlook.com>

* fix: better condition

Signed-off-by: Michael Todorovic <michael.todorovic@outlook.com>

* fix: fix PGStatCheckpointerCollector tests

Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>

---------

Signed-off-by: Michael Todorovic <michael.todorovic@outlook.com>
Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>
Co-authored-by: Michael Todorovic <michael.todorovic@outlook.com>

sysadmind committed:
Prep for v0.17 (#1127)

Signed-off-by: Joe Adams <github@joeadams.io>

vancwo committed:
Fix: Handle incoming labels with invalid UTF-8 (#1131)

It's possible that incoming labels will contain invalid UTF-8 characters. This results in a panic. This fix sanitizes the label's string to ensure only valid UTF-8 characters are included, by replacing invalid characters with � (REPLACEMENT CHARACTER)

Signed-off-by: Cooper Worobetz <cooper@worobetz.ca>

SuperQ committed:
Release v0.17.1 (#1132)

* [BUGFIX] Fix: Handle incoming labels with invalid UTF-8 #1131

Signed-off-by: SuperQ <superq@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.20.5 to 1.21.0 (#1133)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.20.5 to 1.21.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.20.5...v1.21.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#1137)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

dependabot[bot] committed:
Bump golang.org/x/net from 0.33.0 to 0.36.0 (#1138)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.33.0 to 0.36.0.
- [Commits](https://github.com/golang/net/compare/v0.33.0...v0.36.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#1140)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#1142)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

ianbibby committed:
Adds pg_stat_progress_vacuum collector (#1141)

Signed-off-by: Ian Bibby <ian.bibby@reddit.com>
Co-authored-by: Ben Kochie <superq@gmail.com>

SuperQ committed:
Update Go (#1147)

* Update Go to 1.24.
* Update golangci-lint to v2.
* Fixup linting issues.

Signed-off-by: SuperQ <superq@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.21.0 to 1.21.1 (#1144)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.21.0 to 1.21.1.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.21.0...v1.21.1)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-version: 1.21.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.62.0 to 0.63.0 (#1143)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.62.0 to 0.63.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Changelog](https://github.com/prometheus/common/blob/main/RELEASE.md)
- [Commits](https://github.com/prometheus/common/compare/v0.62.0...v0.63.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-version: 0.63.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Sticksman committed:
Record table only size bytes as well in addition to the total size bytes (#1149)

* Record table only size bytes as well in addition to the total size bytes

Signed-off-by: Felix Yuan <felix.yuan@reddit.com>

* Update collector/pg_stat_user_tables.go

Co-authored-by: Ben Kochie <superq@gmail.com>
Signed-off-by: Felix Yuan <felix.yuan@reddit.com>

* Update collector/pg_stat_user_tables.go

Co-authored-by: Ben Kochie <superq@gmail.com>
Signed-off-by: Felix Yuan <felix.yuan@reddit.com>

* Finish renaming elements to index and table size

Signed-off-by: Felix Yuan <felix.yuan@reddit.com>

---------

Signed-off-by: Felix Yuan <felix.yuan@reddit.com>
Co-authored-by: Ben Kochie <superq@gmail.com>

dependabot[bot] committed:
Bump golang.org/x/net from 0.36.0 to 0.38.0 (#1152)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.36.0 to 0.38.0.
- [Commits](https://github.com/golang/net/compare/v0.36.0...v0.38.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-version: 0.38.0
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#1155)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.21.1 to 1.22.0 (#1154)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.21.1 to 1.22.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.21.1...v1.22.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-version: 1.22.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Delorien84 committed:
Export query itself together with queryId in stat_statement metrics (#940)

* Export query itself together with queryId in stat_statement metrics
* The feature must be enabled via flag.
* Limit length of selected query.

The query is not added to every metrics, but instead of new metric stat_statement_query_id is introduced that contains mapping between queryId and query.

Fixes: #813

---------

Signed-off-by: Jakub Štiller <stiller@2n.cz>
Signed-off-by: Jakub Štiller <jakstiller@atlas.cz>

prombot committed:
Update common Prometheus files (#1159)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

BoweFlex committed:
Feat: Improve Error Handling for Server.Scrape (#1158)

* Update querySettings Error Return in Scrape

If there are errors querying namespace mappings, the potential error from querySettings is obscured. Adding an immediate return if there are errors retreiving settings.

Signed-off-by: Jonathan Bowe <jonathan@bowedev.com>

* Improve Verbosity of queryNamespaceMappings Errors

Previously if any errors were encountered by queryNamespaceMappings, only a count of those errors was returned - making debugging those errors harder than it needs to be.

I'm changing this to immediately return nil if no errors are encountered, and otherwise an error will be formatted with each of the namespaces and what the error was for that namespace.

Signed-off-by: Jonathan Bowe <jonathan@bowedev.com>

* Simplify Error Message

Co-authored-by: Ben Kochie <superq@gmail.com>
Signed-off-by: Jonathan Bowe <bowejonathan99@gmail.com>

---------

Signed-off-by: Jonathan Bowe <jonathan@bowedev.com>
Signed-off-by: Jonathan Bowe <bowejonathan99@gmail.com>
Co-authored-by: Ben Kochie <superq@gmail.com>

sfc-gh-pnuttall committed:
Add a collector for `pg_buffercache_summary`. (#1165)

* Add a collector for pg_buffercache_summary()
* Requires PostgreSQL >= 16.
---------

Signed-off-by: Peter Nuttall <peter.nuttall@snowflake.com>
Co-authored-by: Ben Kochie <superq@gmail.com>

prombot committed:
Update common Prometheus files (#1160)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.63.0 to 0.64.0 (#1163)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.63.0 to 0.64.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Changelog](https://github.com/prometheus/common/blob/main/RELEASE.md)
- [Commits](https://github.com/prometheus/common/compare/v0.63.0...v0.64.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-version: 0.64.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.64.0 to 0.65.0 (#1173)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.64.0 to 0.65.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Changelog](https://github.com/prometheus/common/blob/main/RELEASE.md)
- [Commits](https://github.com/prometheus/common/compare/v0.64.0...v0.65.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-version: 0.65.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#1174)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

cristiangreco committed:
chore: fix a typo and use `slices.Contains` (#1176)

Minor fixes as I was skimming through:
- fix typo in `stat_statements`
- drop custom `sliceContains` in `database`

Signed-off-by: Cristian Greco <cristian@regolo.cc>

prombot committed:
Update common Prometheus files (#1178)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

cristiangreco committed:
Replace another custom implementation of `slices.Contains` (#1180)

Followup of https://github.com/prometheus-community/postgres_exporter/pull/1176

Signed-off-by: Cristian Greco <cristian@regolo.cc>

cristiangreco committed:
Ensure database connections are always closed (#1177)

Signed-off-by: Sam DeHaan <sam.dehaan@grafana.com>
Signed-off-by: Sam DeHaan <dehaansa@gmail.com>
Co-authored-by: Sam DeHaan <sam.dehaan@grafana.com>
Co-authored-by: Sam DeHaan <dehaansa@gmail.com>

cristiangreco committed:
Update mixin to latest changes from grafana/postgres_exporter (#1179)

* Update mixin to latest changes from grafana/postgres_exporter

Porting a bunch of PRs accumulated over time in grafana/postgres_exporter:

- https://github.com/grafana/postgres_exporter/pull/11 (@v-zhuravlev)
- https://github.com/grafana/postgres_exporter/pull/12 (@gaantunes)
- https://github.com/grafana/postgres_exporter/pull/13 (@gaantunes)
- https://github.com/grafana/postgres_exporter/pull/14 (@gaantunes)
- https://github.com/grafana/postgres_exporter/pull/15 (@gaantunes)
- https://github.com/grafana/postgres_exporter/pull/16 (@gaantunes)
- https://github.com/grafana/postgres_exporter/pull/17 (@gaantunes)
- https://github.com/grafana/postgres_exporter/pull/20 (@gaantunes)
- https://github.com/grafana/postgres_exporter/pull/21 (@mshahzeb)
- https://github.com/grafana/postgres_exporter/pull/22 (@mshahzeb)

Signed-off-by: Cristian Greco <cristian@regolo.cc>

* rename dashboard to old name

Signed-off-by: Cristian Greco <cristian@regolo.cc>

* remove custom selector

Signed-off-by: Cristian Greco <cristian@regolo.cc>

---------

Signed-off-by: Cristian Greco <cristian@regolo.cc>
Co-authored-by: Vitaly <v-zhuravlev@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.22.0 to 1.23.0 (#1183)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.22.0 to 1.23.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.22.0...v1.23.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-version: 1.23.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kmoppel committed:
Exclude the metrics fetching session's data from pg_stat_activity (#1185)

To reduce the observer effect, filter out pg_stat_activity rows of the
postgres_exporter session from all SA queries, based on pid / procpid.

A bit annoying to see idling DBs showing pg_stat_activity_count "active"
count of 1

Signed-off-by: Kaarel Moppel <kaarel.moppel@gmail.com>

prombot committed:
Update common Prometheus files (#1191)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#1194)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

cristiangreco committed:
Bump prometheus deps (#1195)

-	Update github.com/prometheus/client_golang to v1.23.1
-	Update github.com/prometheus/common to v0.66.1
-	Update github.com/prometheus/exporter-toolkit to v0.14.1

Signed-off-by: Cristian Greco <cristian@regolo.cc>

cristiangreco committed:
Prepare release `0.18.0` (#1196)

* Prepare release `0.18.0`

* [FEATURE] Add `stat_progress_vacuum` collector by @ianbibby
* [FEATURE] Add `buffercache_summary` collector by @sfc-gh-pnuttall
* [FEATURE] `stat_statements`: export query itself together with `queryId` by @Delorien84
* [ENHANCEMENT] Update Go version by @SuperQ
* [ENHANCEMENT] Improve error handling for `Server.Scrape` by @BoweFlex
* [ENHANCEMENT] `stat_user_tables`: record table-only size bytes in addition to the total size bytes by @Sticksman
* [ENHANCEMENT] (chore) Fix a typo and use `slices.Contains` by @cristiangreco
* [ENHANCEMENT] Update mixin to latest changes from `grafana/postgres_exporter` by @cristiangreco, @gaantunes, @v-zhuravlev and @mshahzeb
* [ENHANCEMENT] Exclude the metrics fetching session's data from pg_stat_activity by @kmoppel
* [BUGFIX] Ensure database connections are always closed by @cristiangreco and @dehaansa

Signed-off-by: Cristian Greco <cristian@regolo.cc>

* rm entry about go update

Co-authored-by: Joe Adams <github@joeadams.io>
Signed-off-by: Cristian Greco <cristian@regolo.cc>

* add PRs

Signed-off-by: Cristian Greco <cristian@regolo.cc>

* update date

Signed-off-by: Cristian Greco <cristian@regolo.cc>

---------

Signed-off-by: Cristian Greco <cristian@regolo.cc>
Co-authored-by: Joe Adams <github@joeadams.io>

spantaleev committed:
Fix swapped `flushedLsn` and `receiveStartTli` for `wal_receiver` collector (#1198)

In `pgStatWalReceiverQueryTemplate`, the order of the columns (when `hasFlushedLSN == true`) is:

- ...
- `receive_start_lsn`
- `flushed_lsn`
- `receive_start_tli`
- ...

However, columns were scanned in this order:

- ...
- `receive_start_lsn` -> `receiveStartLsn`
- `receive_start_tli` -> `flushedLsn` (!)
- `flushed_lsn` -> `receiveStartTli` (!)
- ...

This incorrect hydration of variables also manifests as swapped values for the
`pg_stat_wal_receiver_flushed_lsn` and `pg_stat_wal_receiver_receive_start_tli` metrics.

This seems to be a bug that has existed since the initial implementation:

- 2d7e15275147ce7dfb7b7dc6f959d60e106c11a1
- https://github.com/prometheus-community/postgres_exporter/pull/844

In this patch, I'm:

- fixing the `.Scan()`, so that it hydrates variables in the correct order

- adjusting the order in which metrics are pushed out to the channel,
  to follow the order we consume them in
  (.., `receive_start_lsn`, `flushed_lsn`, `receive_start_tli`, ..)

- adjusting the walreceiver tests, to follow the new order (which matches .`Scan()`)

- fixing a small identation issue in `pgStatWalReceiverQueryTemplate`

Signed-off-by: Slavi Pantaleev <slavi@devture.com>

cristiangreco committed:
Fix #1185 bug (#1197) (#1201)

This semicolon breaks the query

Signed-off-by: Joe Adams <github@joeadams.io>
Co-authored-by: Joe Adams <github@joeadams.io>

cristiangreco committed:
Prepare release v0.18.1 (#1202)

* [BUGFIX] Fix swapped `flushedLsn` and `receiveStartTli` for `wal_receiver` collector by @spantaleev in https://github.com/prometheus-community/postgres_exporter/pull/1198
* [BUGFIX] Fix superfluous semicolon breaking query in `process_idle` by @sysadmind in https://github.com/prometheus-community/postgres_exporter/pull/1197 and https://github.com/prometheus-community/postgres_exporter/pull/1201

Signed-off-by: Cristian Greco <cristian@regolo.cc>

SuperQ committed:
Update Go (#1203)

* Update minimum supported Go to 1.24.0.
* Update Go build to 1.25.x.
* Update PostgreSQL testing versions.

Signed-off-by: SuperQ <superq@gmail.com>

cristiangreco committed:
feat: allow setting `limit` in `pg_stat_statements` (#1205)

Add `.limit` CLI flag to `stat_statements` collector to allow
setting a custom number of queries to be returned.

Signed-off-by: Cristian Greco <cristian@regolo.cc>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.23.1 to 1.23.2 (#1204)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.23.1 to 1.23.2.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.23.1...v1.23.2)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-version: 1.23.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

SuperQ committed:
Enable pprof (#1212)

Add the pprof package to allow debug profiling.

Related: https://github.com/prometheus-community/postgres_exporter/issues/1189

Signed-off-by: SuperQ <superq@gmail.com>

prombot committed:
Update common Prometheus files (#1213)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.66.1 to 0.67.2 (#1216)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.66.1 to 0.67.2.
- [Release notes](https://github.com/prometheus/common/releases)
- [Changelog](https://github.com/prometheus/common/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/common/compare/v0.66.1...v0.67.2)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-version: 0.67.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.14.1 to 0.15.0 (#1215)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.14.1 to 0.15.0.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.14.1...v0.15.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-version: 0.15.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Synchronize common files from prometheus/prometheus (#1217)

* Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

* Fix linting issues.

Signed-off-by: SuperQ <superq@gmail.com>

---------

Signed-off-by: prombot <prometheus-team@googlegroups.com>
Signed-off-by: SuperQ <superq@gmail.com>
Co-authored-by: SuperQ <superq@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.67.2 to 0.67.4 (#1224)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.67.2 to 0.67.4.
- [Release notes](https://github.com/prometheus/common/releases)
- [Changelog](https://github.com/prometheus/common/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/common/compare/v0.67.2...v0.67.4)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-version: 0.67.4
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump golang.org/x/crypto from 0.43.0 to 0.45.0 (#1220)

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.43.0 to 0.45.0.
- [Commits](https://github.com/golang/crypto/compare/v0.43.0...v0.45.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-version: 0.45.0
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sysadmind committed:
Update GitHub Actions to run all CI (#1231)

* Update GitHub Actions config to run tests

Signed-off-by: Joe Adams <github@joeadams.io>

* Add CI steps for release and include all postgres versions

Signed-off-by: Joe Adams <github@joeadams.io>

* Disable circleci jobs

This will stop the CI jobs but leave existing PRs in place until we can get those updated.

Signed-off-by: Joe Adams <github@joeadams.io>

---------

Signed-off-by: Joe Adams <github@joeadams.io>

sysadmind committed:
Fix NULL on long_running_transactions collector (#1230)

When there are no long running transactions, the oldest timestamp will be NULL. This sets the metrics value to 0 (age). Also adds a test for this behavior.

Fixes #1223

Signed-off-by: Joe Adams <github@joeadams.io>

prombot committed:
Update common Prometheus files (#1222)

Signed-off-by: prombot <prometheus-team@googlegroups.com>
Co-authored-by: Cristian Greco <cristian@regolo.cc>

avances123 committed:
Document stat_checkpointer collector option (#1226)

Add documentation for stat_checkpointer collector

Signed-off-by: Fabio Rueda <avances123@gmail.com>

prombot committed:
Update common Prometheus files (#1233)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

pierresouchay committed:
Ensure collection of stats succeed in a maximum duration to avoid exhausting PG connections (#1229)

* Ensure collection returns within a specified delay

**Context**: As explained in https://github.com/prometheus-community/postgres_exporter/issues/1228, when database is very slow to answer, every call to prometheus exporter might consume a new connection and possibly consume all available connections

**Solution**: Ensure a Context with a specified Timeout is specified, so the connection will end if duration of collection is too long

Signed-off-by: Pierre Souchay <pierre.souchay@pelico.io>

* test: added TestPGDatabaseTimeout to ensure our timing behaves as it should

Signed-off-by: Pierre Souchay <pierre.souchay@pelico.io>

* document the new variable `PG_EXPORTER_COLLECTION_TIMEOUT` in the README.md

Signed-off-by: Pierre SOUCHAY <pierre.souchay@pelico.io>

* CollectionTimeout -> WithCollectionTimeout

Signed-off-by: Pierre SOUCHAY <pierre.souchay@pelico.io>

* Document and report 0 value for PG_EXPORTER_COLLECTION_TIMEOUT

Signed-off-by: Pierre SOUCHAY <pierre.souchay@pelico.io>

* move test to the right place

Signed-off-by: Pierre SOUCHAY <pierre.souchay@pelico.io>

* ensure that UTest do not take too much time

Signed-off-by: Pierre SOUCHAY <pierre.souchay@pelico.io>

---------

Signed-off-by: Pierre Souchay <pierre.souchay@pelico.io>
Signed-off-by: Pierre SOUCHAY <pierre.souchay@pelico.io>
Co-authored-by: Pierre SOUCHAY <pierre.souchay@pelico.io>

prombot committed:
Update common Prometheus files (#1234)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#1236)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#1237)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

cristiangreco committed:
feat: allow excluding databases or users names in `stat_statements` (#1232)

* feat: allow excluding databases or users names in `stat_statements`

Add a couple of of options to the `stat_statements` collector that
allow excluding certain database names or user names from metrics.

This is useful if you don't care about certain preconfigured databases
with admin activity.

Signed-off-by: Cristian Greco <cristian@regolo.cc>

* address review feedback

Signed-off-by: Cristian Greco <cristian@regolo.cc>

---------

Signed-off-by: Cristian Greco <cristian@regolo.cc>

cristiangreco committed:
ci: add postgres 18.1 to test matrix (#1239)

Signed-off-by: Cristian Greco <cristian@regolo.cc>

prombot committed:
Update common Prometheus files (#1243)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#1244)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

cristiangreco committed:
feat: reorganize code to remove the main package (#1238)

* feat: reorganize code to remove the main package

Move all `main` code to `exporter`, for easier code embedding.

Signed-off-by: Cristian Greco <cristian@regolo.cc>

* move logger args to NewExporter signature

Signed-off-by: Cristian Greco <cristian@regolo.cc>

---------

Signed-off-by: Cristian Greco <cristian@regolo.cc>

prombot committed:
Update common Prometheus files (#1251)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

sysadmind committed:
fix: Do not crash on bad pg_settings value (#1252)

* fix: Do not crash on bad pg_settings value

pg_settings metrics would panic when the parsing fails. This is a poor experience for users. An exporter should not panic. This skips unparsable metrics and instead logs a warning.

Fixes: #1240

---------

Signed-off-by: Joe Adams <github@joeadams.io>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.67.4 to 0.67.5 (#1255)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.67.4 to 0.67.5.
- [Release notes](https://github.com/prometheus/common/releases)
- [Changelog](https://github.com/prometheus/common/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/common/compare/v0.67.4...v0.67.5)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-version: 0.67.5
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.15.0 to 0.15.1 (#1254)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.15.0 to 0.15.1.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.15.0...v0.15.1)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-version: 0.15.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

cristiangreco committed:
Prepare release 0.19.0 (#1256)

Signed-off-by: Cristian Greco <cristian@regolo.cc>

prombot committed:
Update common Prometheus files (#1260)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

cristiangreco committed:
fix: ignore setting `google_dataplex.max_messages` (#1261)

* fix: ignore setting `google_dataplex.max_messages`

Followup of https://github.com/prometheus-community/postgres_exporter/pull/1252

Ignore this setting altogether, as logging a warn at each scrape
for a cloud provider specific entry seems a waste of resources.

Signed-off-by: Cristian Greco <cristian@regolo.cc>

* add comment

Signed-off-by: Cristian Greco <cristian@regolo.cc>

---------

Signed-off-by: Cristian Greco <cristian@regolo.cc>

dannotripp committed:
wal: Fix collector NULL SUM(size) (#1265)

* Fix wal collector NULL SUM(size)
* Use sql.NullInt64 for WAL size, skip metric when NULL

Ref: prometheus-community/postgres_exporter#1264

---------

Signed-off-by: Danno Tripp <danno.tripp@reddit.com>

Nudin committed:
Filter and warn about duplicates in pg_stat_statements (#998) (#1259)

* Filter and warn about duplicates in pg_stat_statements

Signed-off-by: Michael Schönitzer <schoenitzer@pretix.eu>
Signed-off-by: Michael F. Schönitzer <michael@schoenitzer.de>

* Structured log message

Co-authored-by: Cristian Greco <cristian@regolo.cc>
Signed-off-by: Michael F. Schönitzer <michael@schoenitzer.de>

* Use struct instead of string for map of seen objects

Signed-off-by: Michael F. Schönitzer <michael@schoenitzer.de>

* Switch to a map(string)struct{} data structure

Signed-off-by: Michael F. Schönitzer <michael@schoenitzer.de>

---------

Signed-off-by: Michael Schönitzer <schoenitzer@pretix.eu>
Signed-off-by: Michael F. Schönitzer <michael@schoenitzer.de>
Co-authored-by: Cristian Greco <cristian@regolo.cc>

eugercek committed:
Optimize pg_stat_statements queries (#1267)

Signed-off-by: Emin Umut Gercek <eumutgercek@gmail.com>

prombot committed:
Update common Prometheus files (#1266)

Signed-off-by: prombot <prometheus-team@googlegroups.com>
Co-authored-by: Cristian Greco <cristian@regolo.cc>

cristiangreco committed:
Prepare release v0.19.1 (#1268)

Signed-off-by: Cristian Greco <cristian@regolo.cc>

prombot committed:
Update common Prometheus files (#1269)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#1272)

Signed-off-by: prombot <prometheus-team@googlegroups.com>
